rs/compression/src/elias_fano/ef.rs

```diff
@@ -92,66 +136,58 @@ impl EliasFano {
         Ok(())
     }
 
-    /// Returns the value at the given index
-    #[allow(dead_code)]
-    fn get(&self, index: usize) -> Result<u64> {
-        if index >= self.size {
-            return Err(anyhow!("Index {} out of bound", index));
-        }
+    fn len(&self) -> usize {
+        self.size
+    }
 
-        // Calculate the position in upper bits
-        let mut high = 0;
-        let mut pos = 0;
+    fn write(&self, writer: &mut BufWriter<&mut File>) -> Result<usize> {
+        let mut total_bytes_written =
+            wrap_write(writer, &((self.lower_bit_length as u64).to_le_bytes()))?;
+        let lower_vec: &[u64] = self.lower_bits.as_raw_slice();
+        let upper_vec: &[u64] = self.upper_bits.as_raw_slice();
+        total_bytes_written += wrap_write(writer, &((lower_vec.len() as u64).to_le_bytes()))?;
+        total_bytes_written += wrap_write(writer, &((upper_vec.len() as u64).to_le_bytes()))?;
 
-        // Calculate the high part of the value
-        for _ in 0..index + 1 {
-            while pos < self.upper_bits.len() && !self.upper_bits[pos] {
-                // Add the gap to high
-                high += 1;
-                pos += 1;
-            }
-            // Skip the '1' that terminates the unary code
-            pos += 1;
+        for &val in lower_vec.iter() {
+            total_bytes_written += wrap_write(writer, &val.to_le_bytes())?;
         }
-
-        // Calculate the low part of the value
-        let mut low = 0;
-        if self.lower_bit_length > 0 {
-            let low_start = index * self.lower_bit_length;
-            low = (self.lower_bits[low_start..low_start + self.lower_bit_length].load::<u64>()
-                & self.lower_bit_mask) as usize;
+        for &val in upper_vec.iter() {
+            total_bytes_written += wrap_write(writer, &val.to_le_bytes())?;
         }
 
-        Ok((high << self.lower_bit_length | low) as u64)
-    }
+        writer.flush()?;
 
-    /// Returns the number of elements in the encoded sequence
-    pub fn len(&self) -> usize {
-        self.size as usize
+        Ok(total_bytes_written)
```

Comment:
you'd want bit packing here. you're using 64 bits per lower value, and 64 bits per upper value. it's a waste if you need like 3 bits for lower, and much less for upper. pack them as tightly as possible. feel free to write metadata how many bits needed for lower and upper values.